lib/time.js

```diff
@@ -349,6 +415,219 @@ function CronTime(luxon) {
+					this._checkTimeInSkippedRangeSingleHour(
+						startingMinute,
+						afterJumpingPoint.minute
+					)
+				);
+			} else {
+				// non-round or multi-hour jump. (does not exist in the real world at the time of writing)
+				return this._checkTimeInSkippedRangeMultiHour(
+					startingMinute,
+					afterJumpingPoint.minute,
+					startingHour,
+					afterJumpingPoint.hour
+				);
+			}
+		},
+
+		/**
+		 * Component of checking if a CronJob time existed in a DateTime range skipped by DST.
+		 * This subroutine makes a further assumption that the skipped range is fully contained in one hour,
+		 * and that all other larger units are valid for the job.
+		 *
+		 * for example a jump from 02:00:00 to 02:30:00, but not from 02:00:00 to 03:00:00.
+		 * @see _checkTimeInSkippedRange
+		 *
+		 * This is done by checking if any minute in startMinute - endMinute is valid, excluding endMinute.
+		 * For endMinute, there is only a match if the 0th second is a valid time.
+		 */
+		_checkTimeInSkippedRangeSingleHour: function (startMinute, endMinute) {
+			for (let minute = startMinute; minute < endMinute; ++minute) {
+				if (minute in this.minute) return true;
+			}
+
+			// Unless the very last second of the jump matched, there is no match.
+			return endMinute in this.minute && 0 in this.second;
+		},
+
+		/**
+		 * Component of checking if a CronJob time existed in a DateTime range skipped by DST.
+		 * This subroutine assumes the jump touches at least 2 hours, but the jump does not necessarily fully contain these hours.
+		 *
+		 * @see _checkTimeInSkippedRange
+		 *
+		 * This is done by defining the minutes to check for the first and last hour,
+		 * and checking all 60 minutes for any hours in between them.
+		 *
+		 * If any hour x minute combination is a valid time, true is returned.
+		 * The endMinute x endHour combination is only checked with the 0th second, since the rest would be out of the range.
+		 */
+		_checkTimeInSkippedRangeMultiHour: function (
+			startMinute,
+			endMinute,
+			startHour,
+			endHour
+		) {
+			if (startHour >= endHour) {
+				throw new Error(
+					`ERROR: This DST checking related function assumes the forward jump starting hour (${startHour}) is less than the end hour (${endHour})`
+				);
+			}
+
+			/** @type number[] */
+			const firstHourMinuteRange = Array.from(
+				{ length: 60 - startMinute },
+				(_, k) => startMinute + k
+			);
+			/** @type {number[]} The final minute is not contained on purpose. Every minute in this range represents one for which any second is valid. */
+			const lastHourMinuteRange = Array.from(
+				{ length: endMinute },
+				(_, k) => k
+			);
+			/** @type number[] */
+			const middleHourMinuteRange = Array.from({ length: 60 }, (_, k) => k);
+
+			/** @type (number) => number[] */
+			const selectRange = forHour =>
+				forHour === startHour
+					? firstHourMinuteRange
+					: forHour === endHour
+					? lastHourMinuteRange
+					: middleHourMinuteRange;
```

Comment:
this double ternary is really hard to understand. could you refactor it using `if` statements?

Reply:
Done! This lambda was trying way too hard to be without `{}` and `return`.